pyvista/core/_validation/validate.py

```diff
@@ -91,69 +233,82 @@ def validate_array(
-    must_have_shape : int | tuple[int, ...] | list[int, tuple[int, ...]], optional
-        :func:`Check <pyvista.core.validation.check.check_has_shape>`
+    must_have_shape : ShapeLike | list[ShapeLike], optional
+        :func:`Check <pyvista.core._validation.check.check_shape>`
         if the array has a specific shape. Specify a single shape
         or a ``list`` of any allowable shapes. If an integer, the array must
         be 1-dimensional with that length. Use a value of ``-1`` for any
         dimension where its size is allowed to vary. Use ``()`` to allow
         scalar values (i.e. 0-dimensional). Set to ``None`` if the array
         can have any shape (default).
 
-    must_have_dtype : dtype_like | list[dtype_like, ...], optional
-        :func:`Check <pyvista.core.validation.check.check_subdtype>`
+    must_have_dtype : numpy.typing.DTypeLike | Sequence[numpy.typing.DTypeLike], optional
+        :func:`Check <pyvista.core._validation.check.check_subdtype>`
         if the array's data-type has the given dtype. Specify a
-        :class:`np.dtype` object or dtype-like base class which the
-        array's data must be a subtype of. If a ``list``, the array's data
+        :class:`numpy.dtype` object or dtype-like base class which the
+        array's data must be a subtype of. If a sequence, the array's data
         must be a subtype of at least one of the specified dtypes.
 
-    must_have_length : int | array_like[int, ...], optional
-        :func:`Check <pyvista.core.validation.check.check_has_length>`
+    must_have_length : int | Vector[int], optional
+        :func:`Check <pyvista.core._validation.check.check_length>`
         if the array has the given length. If multiple values are given,
         the array's length must match one of the values.
 
         .. note ::
 
             The array's length is determined after reshaping the array
-            (if ``reshape`` is not ``None``) and after broadcasting (if
-            ``broadcast_to`` is not ``None``). Therefore, the values of
-            `length`` should take the array's new shape into
-            consideration if applicable.
+            (if ``reshape_to`` is not ``None``) and after broadcasting (if
+            ``broadcast_to`` is not ``None``). Therefore, the specified length
+            values should take the array's new shape into consideration if
+            applicable.
 
     must_have_min_length : int, optional
-        :func:`Check <pyvista.core.validation.check.check_has_length>`
-        if the array's length is this value or greater.
+        :func:`Check <pyvista.core._validation.check.check_length>`
+        if the array's length is this value or greater. See note in
+        ``must_have_length`` for details.
 
     must_have_max_length : int, optional
-        :func:`Check <pyvista.core.validation.check.check_has_length>`
-        if the array' length is this value or less.
+        :func:`Check <pyvista.core._validation.check.check_length>`
```

Comment:
All the cross-refs in the docs still need to be reviewed to make sure the links work with no typos

Reply:
The docs can be checked as part of #5664